src/robot.cpp

```diff
@@ -959,37 +999,76 @@ void Robot::goalNavigation(const geometry_msgs::PoseStamped& target_pose)
   {
     throw TEMOTO_ERRSTACK("Could not navigate the robot because robot is not operational");
   }
-
   FeatureNavigation& ftr = config_->getFeatureNavigation();
-  std::string act_rob_ns = "/" + config_->getAbsRobotNamespace() + "/move_base";
-  MoveBaseClient ac(act_rob_ns, true);
-  
-  if (!ac.waitForServer(ros::Duration(5.0)))
+  RmNavigationRequestWrap request;
+  request.robot_name = config_->getName();
+  request.goal_pose.header.frame_id = target_pose.header.frame_id;
+  request.goal_pose.pose.position.x = target_pose.pose.position.x;
+  request.goal_pose.pose.position.y = target_pose.pose.position.y;
+  request.goal_pose.pose.position.z = target_pose.pose.position.z;
+  request.goal_pose.pose.orientation.x = target_pose.pose.orientation.x;
+  request.goal_pose.pose.orientation.y = target_pose.pose.orientation.y;
+  request.goal_pose.pose.orientation.z = target_pose.pose.orientation.z;
+  request.goal_pose.pose.orientation.w = target_pose.pose.orientation.w;
+  navigation_feature_plugin_->sendGoal(request);
+
+  /*
+  * Start the navigation feature feedback thread
+  */
+  if (!navigation_feature_feedback_thread_running_)
   {
-    TEMOTO_ERRSTACK("The move_base action server did not come up");
-  }
+    navigation_feature_feedback_thread_running_ = true;
+    navigation_feature_feedback_thread_ = std::thread(
+    [&]
+    {
+      TEMOTO_DEBUG_("Navigation feature feedback thread running");
 
-  move_base_msgs::MoveBaseGoal goal;
-  goal.target_pose = target_pose;
-  goal.target_pose.header.stamp = ros::Time::now();
-  ac.sendGoal(goal);
+      while (navigation_feature_feedback_thread_running_)
+      {
+        std::lock_guard<std::mutex> l(navigation_feature_plugins_mutex_);
+
+        navigation_feature_plugin_->sendUpdate();
+        std::this_thread::sleep_for(std::chrono::milliseconds(200));
+      }
 
-  // Wait until either the goal is finished or robot has encountered a system issue
-  while((ac.getState() == actionlib::SimpleClientGoalState::PENDING || ac.getState() == actionlib::SimpleClientGoalState::ACTIVE)
+      TEMOTO_DEBUG_("Navigation feature feedback thread finished");
+    });
+  }
+
+  while((navigation_feature_plugin_->getState() == temoto_robot_manager::NavigationPluginHelper::State::PROCESSING)
```

Comment:
this is the reason why you had to comment out the mutex in the `robot_manager.cpp` line 791. you shouldn't be waiting here. just send the goal and return. refer to the implementaiton of `invokeCustomFeature`

Reply:
mutex back. 
while loop moved to RMI